alacritty/src/input.rs

```diff
@@ -1192,6 +1243,251 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
+        Key::MediaFastForward => ("57433".into(), 'u'),
+        Key::MediaRewind => ("57434".into(), 'u'),
+        Key::MediaTrackNext => ("57435".into(), 'u'),
+        Key::MediaTrackPrevious => ("57436".into(), 'u'),
+        Key::MediaRecord => ("57437".into(), 'u'),
+        Key::AudioVolumeDown => ("57438".into(), 'u'),
+        Key::AudioVolumeUp => ("57439".into(), 'u'),
+        Key::AudioVolumeMute => ("57440".into(), 'u'),
+        Key::Escape if named_csi_u => ("27".into(), 'u'),
+        // Keys which are reported only when all key must be reported
+        Key::CapsLock if encode_all => ("57358".into(), 'u'),
+        Key::NumLock if encode_all => ("57360".into(), 'u'),
+        // Left mods.
+        Key::Shift if key.location == KeyLocation::Left && encode_all => ("57441".into(), 'u'),
+        Key::Control if key.location == KeyLocation::Left && encode_all => ("57442".into(), 'u'),
+        Key::Alt if key.location == KeyLocation::Left && encode_all => ("57443".into(), 'u'),
+        Key::Super if key.location == KeyLocation::Left && encode_all => ("57444".into(), 'u'),
+        Key::Hyper if key.location == KeyLocation::Left && encode_all => ("57445".into(), 'u'),
+        Key::Meta if key.location == KeyLocation::Left && encode_all => ("57446".into(), 'u'),
+        // Right mods.
+        Key::Shift if key.location == KeyLocation::Right && encode_all => ("57447".into(), 'u'),
+        Key::Control if key.location == KeyLocation::Right && encode_all => ("57448".into(), 'u'),
+        Key::Alt if key.location == KeyLocation::Right && encode_all => ("57449".into(), 'u'),
+        Key::Super if key.location == KeyLocation::Right && encode_all => ("57450".into(), 'u'),
+        Key::Hyper if key.location == KeyLocation::Right && encode_all => ("57451".into(), 'u'),
+        Key::Meta if key.location == KeyLocation::Right && encode_all => ("57452".into(), 'u'),
+
+        Key::Enter if encode_all => ("13".into(), 'u'),
+        Key::Tab if encode_all => ("9".into(), 'u'),
+        Key::Backspace if encode_all => ("127".into(), 'u'),
+        // When the character key ended up being a text, like when compose was done.
+        Key::Character(c) if encode_all && c.chars().count() > 1 => ("0".into(), 'u'),
+        Key::Character(c) => {
+            let character = c.chars().next().unwrap();
+            let base_character = character.to_lowercase().next().unwrap();
+
+            let codepoint = u32::from(character);
+            let base_codepoint = u32::from(base_character);
+
+            // NOTE: Base layouts are ignored, since winit doesn't expose this information.
+            let payload = if mode.contains(TermMode::KEYBOARD_REPORT_ALTERNATE_KEYS)
+                && codepoint != base_codepoint
+            {
+                format!("{codepoint}:{base_codepoint}")
+            } else {
+                codepoint.to_string()
+            };
+
+            (payload.into(), 'u')
+        },
```

Comment:
Having such a big code block in such an already complex match statement is unacceptable.